### PR TITLE
added __getitem__ for iterable behavior

### PR DESCRIPTION
**Pull request by s-light**
i added __getitem__ and __len__ magic functions to the CRGB and CHSV classes.
this way the get iterable and you can access the color-parts in list style.
so for other classes its easier to access the content in the same form as a plain list or tuple.

let me know what you think of this.

this idea came to my mind as i implemented the [TLC5957 driver](https://github.com/s-light/slight_CircuitPython_TLC5957)

### CODE REVIEW

File: adafruit_fancyled/adafruit_fancyled.py [hunk 1]
```diff
@@ -114,6 +114,21 @@ def __repr__(self):
...
+        """Retrieve total number of color-parts available."""
+        return 3
+
+    def __getitem__(self, key):
+        """Retrieve red, green or blue value as iterable."""
+        if key is 0:
+            return self.red
+        elif key is 1:
+            return self.green
+        elif key is 2:
```
Comment (42B) [inline adafruit_fancyled/adafruit_fancyled.py]: Shouldn't these three be checking for equality and not identity? `is` -> `==`

File: adafruit_fancyled/adafruit_fancyled.py [hunk 2]
```diff
@@ -164,6 +179,21 @@ def __repr__(self):
...
+        """Retrieve total number of 'color-parts' available."""
+        return 3
+
+    def __getitem__(self, key):
+        """Retrieve hue, saturation or value as iterable."""
+        if key is 0:
+            return self.hue
+        elif key is 1:
+            return self.saturation
+        elif key is 2:
```
Comment (42B) [inline adafruit_fancyled/adafruit_fancyled.py]: Equality instead of identity? `is` -> `==`

Comment (s-light) [reply adafruit_fancyled/adafruit_fancyled.py]: good catch - correct ;-)

File: adafruit_fancyled/adafruit_fancyled.py (hunk 1 above)
Comment (s-light) [reply adafruit_fancyled/adafruit_fancyled.py]: good catch - correct ;-)

**Comment by jrowe88** [comment]
Thanks @kattni ! __len__ makes it play nicer with dotstar library.  Is someone going to merge this in?

**Comment by PaintYourDragon** [comment]
Looks good, OK to merge.

**Pull request merged**